static/js/lib/learning_resources.js

```diff
@@ -27,24 +34,131 @@ export const availabilityLabel = (availability: ?string) => {
   }
 }
 
+export const parseDateFilter = (filter: string) => {
```

Comment:
so this function is I think taking the text and returning a moment object for comparing, right? two suggestions:

1. a name like `dateFilterToMoment` or something similar might better capture what the function does
2. we have a fixed set of possible values for the availability filter (in our `AVAILABILITY_MAPPING`) so it might be clearer to have a switch statement on those constant values, like

```js
switch (filter) {
  case AVAILABLE_NOW:
    return moment.now()
  case AVAILABLE_NEXT_WEEK:
    return moment.now().add(7, "days")
```

something along those lines, as-is it's pretty hard to follow what's going on here with the regex and everything.

Reply:
The filters defined in `AVAILABILITY_MAPPING` are also used in `lib/search/buildSearchQuery`, so I'd prefer to also use them here rather than ignore them and replace with a longer switch statement that repeats some of the same logic and would risk getting out of sync with the `AVAILABILITY_MAPPING` filters.  Will definitely rename it though and can add an explanatory comment or two.

Reply:
ah, yeah I didn't think about it getting out of sync with the search logic, that's definitely a possibility. I might suggest then that instead we have a separate function that just takes the availability filter string and returns values that `moment` can use, so that part can be pulled out and be cleanly defined